x/gov/migrations/v046/convert_test.go

```diff
@@ -0,0 +1,175 @@
+package v046_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	v046 "github.com/cosmos/cosmos-sdk/x/gov/migrations/v046"
+	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConvertToLegacyProposal(t *testing.T) {
+	propTime := time.Unix(1e9, 0)
+	legacyContentMsg, err := v1.NewLegacyContent(v1beta1.NewTextProposal("title", "description"), "cosmos1fl48vsnmsdzcv85q5d2q4z5ajdha8yu34mf0eh")
+	require.NoError(t, err)
+	msgs := []sdk.Msg{legacyContentMsg}
+	msgsAny, err := tx.SetMsgs(msgs)
+	require.NoError(t, err)
+	proposal := v1.Proposal{
+		Id:              1,
+		Status:          v1.StatusDepositPeriod,
+		Messages:        msgsAny,
+		SubmitTime:      &propTime,
+		DepositEndTime:  &propTime,
+		VotingStartTime: &propTime,
+		VotingEndTime:   &propTime,
+		Metadata:        "proposal metadata",
+	}
+
+	testCases := map[string]struct {
+		tallyResult v1.TallyResult
+		expErr      bool
+	}{
+		"valid": {
+			tallyResult: v1.EmptyTallyResult(),
+		},
+		"invalid final tally result": {
+			tallyResult: v1.TallyResult{},
+			expErr:      true,
```

Comment:
Why are we erroring when an empty tally result? `ConvertToLegacyProposal` should be able to convert open proposals too, no?

Reply:
when submitting a proposal, the tally result is prefilled with `EmptyTallyResult()`: https://github.com/cosmos/cosmos-sdk/blob/feb9d93d4d607f095ec4494b1e6579ece391f716/x/gov/types/v1/proposal.go#L33 which gives "0" zero values to the tally counts, so I don't think there should be any v1 proposals in state with empty strings in the tally result

Reply:
Ah yes, makes sense!

Reply:
also in v1beta1 proposal tally result, counts are stored as sdk.Int so it can't work if we try to convert some v1 proposal tally result counts with empty string